### PR TITLE
[improve][broker][PIP-149]make examineMessage method async

### PR DESCRIPTION
**Pull request by HQebupt**
Master Issue: #14013 
### Motivation

See #14365 

### Verifying this change

Make sure that the change passes the CI checks.

### Does this pull request potentially affect one of the following parts:

If `yes` was chosen, please highlight the changes

- Dependencies (does it add or upgrade a dependency): (no)
- The public API: (no)
- The schema: (no)
- The default values of configurations: (no)
- The wire protocol: (no)
- The rest endpoints: (no)
- The admin cli options: (no)
- Anything that affects deployment: (no)

### Documentation
Check the box below and label this PR (if you have committer privilege).

Need to update docs? 
- [x] `doc-not-needed`

### CODE REVIEW

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java [hunk 1]
```diff
@@ -2851,58 +2851,77 @@ protected Response internalPeekNthMessage(String subName, int messagePosition, b
...
-                }
-            }, null);
-            return generateResponseWithEntry(future.get());
-        } catch (Exception exception) {
-            exception.printStackTrace();
-            log.error("[{}] Failed to examine message at position {} from {} due to {}", clientAppId(), messagePosition,
-                    topicName, exception);
-            throw new RestException(exception);
-        }
+                }).thenCompose(entry -> {
```
Comment (Technoboy-) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java:2914]: `thenApply` is more better ?

**Review by Technoboy- — APPROVED** [review]

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java [hunk 2]
```diff
@@ -2851,58 +2851,77 @@ protected Response internalPeekNthMessage(String subName, int messagePosition, b
...
+                        } else {
+                            return CompletableFuture.completedFuture(null);
+                        }
+                    });
         }
 
-        if (null == initialPosition) {
-            initialPosition = "latest";
-        }
+        return ret.thenCompose(__ -> validateTopicOwnershipAsync(topicName, authoritative))
```
Comment (gaozhangmin) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java:2878]: `validateTopicOperationAsync` should be preceded the `getPartitionedTopicMetadataAsync`

**Review by gaozhangmin — CHANGES_REQUESTED** [review]

**Review by gaozhangmin — APPROVED** [review]

**Comment by gaozhangmin** [comment]
@Technoboy- PTAL again, changes made.

**Pull request merged**